feat(docs): add live-preview, props, return values to `useTable` (#3439)
refinedev/refine: documentation/docs/api-reference/mui/hooks/useDataGrid/index.md

---
title: useDataGrid
slug: /api-reference/mui/hooks/useDataGrid
source: /packages/mui/src/hooks/useDataGrid
---

import BasicUsagePreview from "./\_partial-use-data-grid-basic-usage-live-preview.md";
import RelationalPreview from "./\_partial-use-data-grid-relational-live-preview.md";
import PropResource from "@site/src/partials/prop-resource";

By using `useDataGrid`, you can get properties that are compatible with MUI X [`<DataGrid>`][data-grid] component. All features such as sorting, filtering, and pagination come out of the box. Under the hood it uses [`useList`](/docs/api-reference/core/hooks/data/useList/) for the fetch.

For all the other features, you can refer to the MUI X [`<DataGrid>`][data-grid] documentation

> > 💡 The `useDataGrid` hook works in compatibility with both the [`<DataGrid>`][data-grid] and the [`<DataGridPro>`](https://mui.com/x/react-data-grid/#commercial-version) component.

Reviewer: ```suggestion
> 💡 The `useDataGrid` hook is compatible with both the [`<DataGrid>`][data-grid] and the [`<DataGridPro>`](https://mui.com/x/react-data-grid/#commercial-version) components.
```
